public/app/features/alerting/unified/components/silences/SilencedInstancesPreview.tsx

```diff
@@ -14,25 +18,43 @@ import { AmAlertStateTag } from './AmAlertStateTag';
 
 interface Props {
   amSourceName: string;
-  matchers: Matcher[];
+  matchers: MatcherFieldValue[];
+  ruleUid?: string;
 }
 
-export const SilencedInstancesPreview = ({ amSourceName, matchers }: Props) => {
-  const { useGetAlertmanagerAlertsQuery } = alertmanagerApi;
+const useDebouncedDeepCompare = (cb: () => void, debounceMs: number, dependencies: unknown[]) => {
+  const [state, setState] = useState<unknown[]>();
+
+  useDebounce(cb, debounceMs, [state]);
+
+  useDeepCompareEffect(() => {
+    setState(dependencies);
+  }, [dependencies]);
+};
+
+export const SilencedInstancesPreview = ({ amSourceName, matchers: inputMatchers, ruleUid }: Props) => {
+  const matchers: Matcher[] = [
```

Comment:
This code creates a new instance of `matchers` array every time so maybe the deep compare later or is not needed?

Reply:
We need the deep compare as we're checking the values - if we don't do that, we end up thinking that they change all the time and we call the API too frequently. I couldn't get it working without this approach, but happy if you're able to get it working